### DISCUSSION TITLE
DatabaseTransactions and mongodb

### DISCUSSION THREAD

**Discussion by ozanuzer** [jenssegers/laravel-mongodb#2732] [General] [answered]
- Laravel-mongodb Version: 4.1.1
- PHP Version: 8.3.3
- Database Driver & Version: php mongodb extension 1.17.2 & MongoDB 7.0.2 Community

### Description:
Hello, I am trying to run unit test and test including inserting and fetch data with mongodb. When I run test I am facing "Transaction numbers are only allowed on a replica set member or mongos". When I googling it, I can see "use DatabaseTransactions;" is problem. How can I fix that? I am using macos.

**Comment by GromNaN** [comment, marked as answer]
By design, MongoDB transactions require a replica set.

Possible solutions:
- For local development, I use [mongo-orchestration](https://github.com/10gen/mongo-orchestration) on MacOS
- You can create a single-instance replicaset: [Convert a Standalone mongod to a Replica Set](https://www.mongodb.com/docs/manual/tutorial/convert-standalone-to-replica-set/)
- If you use Docker, it may be the simplest solution (outdated docker-compose replicaset: https://github.com/UpSync-Dev/docker-compose-mongo-replica-set)
